assets/src/edit-story/components/devTools/devTools.js

```diff
@@ -0,0 +1,202 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useGlobalKeyDownEffect } from '../keyboard';
+import { useStory } from '../../app/story';
+import { useSnackbar } from '../../app/snackbar';
+import { dummyImage, dummyVideo } from './dummyData';
+
+const Container = styled.div`
+  position: fixed;
+  z-index: 2147483646;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  flex-direction: column;
+  width: 35%;
+  height: 50%;
+  border: 6px solid ${({ theme }) => theme.colors.bg.v4};
+  background: ${({ theme }) => theme.colors.bg.v1};
+  color: ${({ theme }) => theme.colors.fg.v2};
+`;
+
+const Textarea = styled.textarea`
+  flex: 1;
+  border: 0;
+  background: ${({ theme }) => theme.colors.bg.v1};
+  color: ${({ theme }) => theme.colors.fg.v2};
+  white-space: nowrap;
+  overflow: auto;
+`;
+
+const Options = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 6px;
+`;
+
+const Label = styled.label`
+  margin-right: 6px;
+`;
+
+const Button = styled.button`
+  margin-left: 6px;
+`;
+
+const replaceResourcesWithDummy = (state) => {
+  return {
+    ...state,
+    pages: state.pages.map((page) => ({
+      ...page,
+      elements: page.elements.map((element) => {
+        let newElement = { ...element };
+        if ('resource' in element) {
+          newElement.resource = { ...element.resource };
+          if (element.type === 'video') {
+            newElement.resource.mimeType = 'video/mp4';
+            newElement.resource.src = dummyVideo;
+            newElement.resource.poster = dummyImage;
+
+            const videoEl = document.getElementById(`video-${newElement.id}`);
+            if (videoEl) {
+              videoEl.load();
+            }
+          }
+          if (element.type === 'image') {
+            newElement.resource.mimeType = 'image/png';
+            newElement.resource.src = dummyImage;
+          }
+        }
+        return newElement;
+      }),
+    })),
+  };
+};
+
+function DevTools() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [isBase64, setIsBase64] = useState(false);
+  const [isDummyResources, setIsDummyResources] = useState(false);
+  const { showSnackbar } = useSnackbar();
+  const textareaRef = useRef();
+  const {
+    internal: { reducerState, restore },
+  } = useStory();
+
+  const {
+    pages,
+    current,
+    selection,
+    story: { stylePresets },
+  } = reducerState;
+  const reducerStateSlice = {
+    current,
+    selection,
+    story: { stylePresets },
+    pages,
+  };
+  const storyData = isDummyResources
+    ? replaceResourcesWithDummy(reducerStateSlice)
```

Comment:
I think you may want `useMemo()` here. Otherwise you will be running replacement logic on each rerender operation.

Reply:
Yes, but at the same time if this useMemo is dependant on story data(`reducerStateSlice`) it will re-compute every time you touch the UI anyway?

Also, we will run `replaceResourcesWithDummy` only if you have `isDummyResources` enabled (that means DevTools open, so just in this specific dev scenario).